src/midcom/datamanager/storage/parameter.php

```diff
@@ -29,6 +46,105 @@ public function save()
             && $this->load() === null) {
             return true;
         }
+        if(isset($this->config['type_config']['multiple_storagemode'])) {
+            $this->value = $this->convert_multiple_to_storage();
+        }
+
         return $this->object->set_parameter($this->config['storage']['domain'], $this->config['storage']['name'], $this->value);
     }
-}
+
+    private function convert_multiple_from_storage($source)
+    {
+        $glue = $this->multiple_separator;
+
+        switch ($this->config['type_config']['multiple_storagemode']) {
+            case 'serialized':
+            case 'array':
+                if (   !is_array($source)
+                    && empty($source)) {
+                    $source = array();
+                }
+                return $source;
+
+            case 'imploded':
+                if (!is_string($source)) {
+                    return array();
+                }
+                return explode($glue, $source);
+
+            case 'imploded_wrapped':
+                if (!is_string($source)) {
+                    return array();
+                }
+                return explode($glue, substr($source, 1, -1));
+
+            default:
+                throw new midcom_error("The multiple_storagemode '{$this->multiple_storagemode}' is invalid, cannot continue.");
+        }
+    }
+
+    /**
+     * Converts the selected options according to the multiple_storagemode setting.
+     *
+     * @return mixed The data converted to the final data storage.
+    */
+    function convert_multiple_to_storage()
+    {
+        switch ($this->config['type_config']['multiple_storagemode']) {
+            case 'array':
+                return $this->value;
+
+            case 'serialized':
+                if ($this->others) {
+                    return array_merge($this->value, $this->others);
+                }
+                return $this->value;
+
+            case 'imploded':
+                $options = $this->getImplodedOptions();
+                return $options;
+
+            case 'imploded_wrapped':
+                $glue = $this->multiple_separator;
+                $options = $this->getImplodedOptions();
+                return "{$glue}{$options}{$glue}";
+
+            default:
+                throw new midcom_error("The multiple_storagemode '{$this->config['type_config']['multiple_storagemode']}' is invalid, cannot continue.");
+        }
+    }
+
+    /**
+     * Prepares the imploded storage string. All entries containing the pipe char (used as glue)
+     * will be logged and skipped silently.
+     *
+     * @return string The imploded data string.
+     */
+    private function getImplodedOptions()
```

Comment:
Pls don't mix naming conventions. the other mehtods in this class all have underscores, only this one uses CamelCase